tools/development/parser/convert.c

```diff
@@ -0,0 +1,147 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/**
+ * GStreamer Pipeline from/to PBTxt Converter Parser
+ * Copyright (C) 2021 MyungJoo Ham <myungjoo.ham@samsung.com>
+ * Copyright (C) 2021 Dongju Chae <dongju.chae@samsung.com>
+ */
+/**
+ * @file    convert.c
+ * @date    13 May 2021
+ * @brief   GStreamer pipeline from/to pbtxt converter
+ * @see     http://github.com/nnstreamer/nnstreamer
+ * @author  Dongju Chae <dongju.chae@samsung.com>
+ * @bug     No known bugs except for NYI items
+ */
+
+#include "convert.h"
+#include <glib/gprintf.h>
+
+GData *datalist;
+
+#define GET_NODE_INDEX(elem)\
+  g_slist_index (g_datalist_get_data (&datalist, elem->element), elem)
+
+/**
+ * @brief Get a node name (ignore index 0)
+ */
+static gchar *
+pbtxt_get_node_name (_Element *elem)
+{
+  gint index = GET_NODE_INDEX (elem);
```

Comment:
What happens if ```GET_NODE_INDEX``` returns -1? Is it impossible?

Reply:
Yes, it's impossible when parsing was successful.

Reply:
I'll add `g_assert` to ensure this.